kedro/framework/cli/cli.py

```diff
@@ -199,6 +213,19 @@ def main(
                 click.echo(message)
                 click.echo(hint)
             sys.exit(exc.code)
+        except Exception as error:
+            logger.error(f"An error has occurred: {error}")
+            exit_code = 1
+            self._cli_hook_manager.hook.after_command_run(
+                project_metadata=self._metadata, command_args=args, exit_code=1
+            )
+            hook_called = True
+        finally:
+            if not hook_called:
```

Comment:
Thank you for the PR, @lrcouto! Could you please explain how this is possible? It looks like `hook_called` will be set to `True` in the try block or, if it fails, in the except block, so this line will never be executed?

Reply:
I guess that's in case of no exception

Reply:
Yes, this finally block is there to solve that issue where the hook was called twice when there was not an exception.